### ISSUE TITLE
Triple filtering strategy broken with BIND in UNION statements

### ISSUE THREAD

**Issue by enridaga** [SPARQL-Anything/sparql.anything#89]
When the query pattern includes a UNION statement no triple is loaded. This works correctly when passing -s 0 .
The problem is that -s 1 is the default on supported transformers such as JSON!

**Comment by enridaga** [comment]
The UNION patterns seem considered properly: https://github.com/SPARQL-Anything/sparql.anything/blob/e23028d1a868203bc96ea13e754435573c1274e7/sparql.anything.model/src/main/java/com/github/spiceh2020/sparql/anything/model/TripleFilteringFacadeXBuilder.java#L333

**Comment by enridaga** [comment]
I reproduced the error and it seems that the problem is not the UNION but the BIND statement. However, it seems happening only when inside a UNION statement! Will prepare a test case.

**Comment by enridaga** [comment]
This the problem is this, instead, where the `opAssign` is not considered by the triple filtering method: https://github.com/SPARQL-Anything/sparql.anything/blob/e23028d1a868203bc96ea13e754435573c1274e7/sparql.anything.model/src/main/java/com/github/spiceh2020/sparql/anything/model/TripleFilteringFacadeXBuilder.java#L311

**Comment by enridaga** [comment]
Precisely, the bug occurred when two UNION clauses included a BIND operator.

To reproduce with released version v0.3.0:

File content:
`
{ "field":true }
`

Query:
```
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>


SELECT *

WHERE {
	SERVICE <x-sparql-anything:> {
	  fx:properties fx:location "./broken-union.json" ;
	    fx:media-type "application/json" 
		.
		{
			?a ?b ?c
			BIND("A" as ?letter)
		} UNION {
			?d ?e ?f
			BIND("D" as ?letter)
		} 

	}
}
```
Running `fx -q queryFile -f CSV should return 4 rows:
```
a,b,c,letter,d,e,f
_:b0,http://www.w3.org/1999/02/22-rdf-syntax-ns#type,http://sparql.xyz/facade-x/ns/root,A,,,
_:b0,http://sparql.xyz/facade-x/data/field,true,A,,,
,,,D,_:b0,http://www.w3.org/1999/02/22-rdf-syntax-ns#type,http://sparql.xyz/facade-x/ns/root
,,,D,_:b0,http://sparql.xyz/facade-x/data/field,true
```
but, with version `v0.3.0`, you get an empty result set.
